### PR TITLE
fix: capture scheduler exception errors in Optimus notifications

### PR DESCRIPTION
**Pull request by Mryashbhardwaj**

### CODE REVIEW

File: ext/scheduler/airflow2/resources/__lib.py [hunk 1]
```diff
@@ -322,17 +326,18 @@ def _parse_datetime(self, timestamp) -> datetime:
...
-def optimus_failure_notify(context, event_meta):
+def optimus_notify(context, event_meta):
     params = context.get("params")
     optimus_client = OptimusAPIClient(params["optimus_hostname"])
 
-    taskfail_alert = int(Variable.get("taskfail_alert", default_var=1))
-    if taskfail_alert != 1:
-        return "suppressed failure alert"
+    # # what to do of this ? 
+    # taskfail_alert = int(Variable.get("taskfail_alert", default_var=1))
```
[the remark this answers is not in the document]
Comment (Mryashbhardwaj) [reply ext/scheduler/airflow2/resources/__lib.py]: resolved

File: job/event.go [hunk 2]
```diff
@@ -34,6 +34,7 @@ func (e *eventService) Register(ctx context.Context, namespace models.NamespaceS
 	evt models.JobEvent) error {
 	var err error
 	for _, notify := range jobSpec.Behavior.Notify {
+		//bug here: with new notification types, this wont work
```
[the remark this answers is not in the document]
Comment (Mryashbhardwaj) [reply job/event.go]: handled ,

File: models/job.go [hunk 3]
```diff
@@ -62,6 +62,10 @@ const (
 	JobRetryEvent JobEventType = "retry"
 )
 
+var faiulreEvents = []JobEventType{
```
Comment (sravankorumilli) [inline models/job.go:65]: we can keep this internal to Should Notify or a method on the struct

File: ext/scheduler/airflow2/resources/__lib.py [hunk 4]
```diff
@@ -261,6 +263,8 @@ def poke(self, context):
                 upstream_schedule = self.get_schedule_interval(schedule_time)
             except Exception as e:
                 self.log.warning("error while fetching upstream schedule :: {}".format(e))
+                context[SCHEDULER_ERR_MSG] = e
```
Comment (sravankorumilli) [inline ext/scheduler/airflow2/resources/__lib.py:266]: good to pass just the error message

**Review by sravankorumilli — CHANGES_REQUESTED** [review]

**Review by sravankorumilli — APPROVED** [review]
Feel free to merge once the changes are done

File: models/job.go (hunk 3 above)
Comment (Mryashbhardwaj) [reply models/job.go:65]: sure

File: ext/scheduler/airflow2/resources/__lib.py (hunk 4 above)
Comment (Mryashbhardwaj) [reply ext/scheduler/airflow2/resources/__lib.py:266]: sure

File: models/job.go (hunk 3 above)
Comment (Mryashbhardwaj) [reply models/job.go:65]: okay

**Pull request merged**